### PR TITLE
Add a move ctor for p::d::Vector.

### PR DESCRIPTION
**Pull request by drwells**

### CODE REVIEW

**Comment by drwells** [comment]
This is harder to implement than I thought - a move needs to leave the moved-from object in a valid state, which means we need to allocate memory (we need an empty partitioner) and thus we cannot be `noexcept`.

File: include/deal.II/lac/la_parallel_vector.h [hunk 1]
```diff
@@ -282,6 +282,15 @@ namespace LinearAlgebra
...
       Vector(const Vector<Number, MemorySpace> &in_vector);
 
+      /**
+       * Move constructor. Uses the swap method.
+       *
+       * @note In order for this constructor to leave the moved-from object in a
+       * valid state it must allocate memory (in this case, an empty
+       * partitioner) - hence it cannot be marked as noexcept.
+       */
+      Vector(Vector<Number, MemorySpace> &&in_vector); // NOLINT
```
Comment (drwells) [inline include/deal.II/lac/la_parallel_vector.h:292]: This is not enough to make clang-tidy happy. I still think it makes sense to define a non-`noexcept` move constructor, so we need to turn that check off for this line. Is there some other syntax for this?

**Review by kronbichler — APPROVED** [review]

File: include/deal.II/lac/la_parallel_vector.h (hunk 1 above)
Comment (drwells) [reply include/deal.II/lac/la_parallel_vector.h:292]: I figured it out - I have to write `// NOLINT` at the end of the line containing the start of the ctor definition in the `.templates.h` file too.

**Pull request merged**